static/app/actionCreators/modal.tsx

```diff
@@ -308,3 +308,16 @@ export async function openWidgetViewerModal({
     onClose,
   });
 }
+
+export async function openCreateNewIntegrationModal(
+  options: CreateNewIntegrationModalOptions
+) {
+  const mod = await import('sentry/components/modals/createNewIntegrationModal');
+  const {default: Modal} = mod;
+
+  openModal(deps => <Modal {...deps} {...options} />);
+}
+
+type CreateNewIntegrationModalOptions = {
+  orgId: string;
+};
```

Comment:
This happens to coordinate with `type Props` declared in `createNewIntegrationModal.tsx` but that means if we change props there, this won't update. You should probably extend it instead with `CreateNewIntegration['props']` or something similar instead.